reinstall-cli/src/users.rs

```diff
@@ -0,0 +1,106 @@
+use anyhow::{Context, Result};
+use std::fmt::Display;
+use std::fmt::Formatter;
+use uzers::os::unix::UserExt;
+
+#[derive(Clone, Debug)]
+pub(crate) struct UserKeys {
+    pub(crate) user: String,
+    pub(crate) authorized_keys: String,
+    pub(crate) authorized_keys_path: String,
+}
+
+impl UserKeys {
+    pub(crate) fn num_keys(&self) -> usize {
+        self.authorized_keys.lines().count()
+    }
+}
+
+impl Display for UserKeys {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "User {} ({} authorized keys)",
+            self.user,
+            self.num_keys()
+        )
+    }
+}
+
+/// # Safety
+/// See safety information in [`uzers::all_users`]. We are not impacted because we're not
+/// multi-threaded.
+pub(crate) fn get_all_users_keys() -> Result<Vec<UserKeys>> {
+    #[allow(unsafe_code)]
+    let iter = unsafe { uzers::all_users() };
```

Comment:
It may be simpler and more reliable here to parse the output of `loginctl -j`. That will only give us logged in users, which in the common case of non-root user + sudo will give us both, and in the case of just logging in as root won't offer other users, which also seems right.

Reply:
I assume you're referring to `loginctl --output=json`?

That doesn't seem to give me any information about root, or the user's home directories, even with `--all`

Reply:
> That doesn't seem to give me any information about root, or the user's home directories, even with --all

I mean that `loginctl` replaces `uzers::all_users()` - we don't actually want to iterate over *all* users (necessarily) just logged in ones. (bonus: we kill an `unsafe`). We'd still use `uzers::get_user_by_name` on the names we find from `loginctl` to gather other data.

As far as `root`, if it doesn't show up from logged in users we could just add it to the list automatically.

Reply:
Alright

Reply:
Seems like the exact flag changed in recent systemd versions  

Related:
https://github.com/kolide/launcher/issues/1758
https://github.com/systemd/systemd/issues/15275
https://github.com/systemd/systemd/pull/16176